todolistfrontend/src/context/Auth.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState, createContext, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import { GetToken } from '../services/storage.services';
+
+export const AuthContext = createContext({
+  isAuthenticated: false,
+  login: () => {},
+  logout: () => {},
+});
+
+export const AuthContextProvider = ({ children }) => {
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+  const token = GetToken();
+
+  useEffect(() => {
+    if (!token) return;
+
+    setIsAuthenticated(true);
+    login(token);
+  }, []);
+
+  useEffect(() => {
+    if (!isAuthenticated) return;
+    history.push('/');
+  }, [isAuthenticated]);
+
+  const history = useHistory();
```

Comment:
When intializing hooks try to keep them in one place. Usually they are init at the top and again with order you decide, (E.g. firstly useState then useHistory useContext etc). useMemo useCallback useEffect and hooks like that are exception.